File: best_time_to_buy_and_sell_stock/best_time_to_buy_and_sell_stock.cpp
class Solution
{
public:
    int maxProfit(vector<int> &prices)
    {
        int buy = 0;
        int sell = 0;
        int maxProfit = 0;
        for (int i = 0; i < prices.size(); i++)
        {
            if (prices[buy] < prices[sell])
            {
                int currentProfit = prices[sell] - prices[buy];
                maxProfit = max(maxProfit, currentProfit);
            }
            else
            {
                buy = sell;
            }
            sell++;
        }
        return maxProfit;
    }
};

File: best_time_to_buy_and_sell_stock/best_time_to_buy_and_sell_stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy, sell = 0, 0
        maxProfits = 0
        n = len(prices)
        while sell < n:
            if prices[buy] < prices[sell]:
                currentProfit = prices[sell] - prices[buy]
                maxProfits = max(currentProfit, maxProfits)
            else:
                buy = sell
            sell += 1
        return maxProfits
